### PR TITLE
Eth transfer

### PR DESCRIPTION
**Pull request by admrid**
#### Steps to reproduce:
go to /transfer and send A€ to an address that does not have ETH -not working yet -WIP

#### Trello card link:
https://trello.com/c/dYF3p7t7

#### Is connection necessary to test? If so which network?
- [ ] local RPC
- [x] Rinkeby
- [x] Main Ethereum Network

### CODE REVIEW

**Comment by szerintedmi** [comment]
Deploy preview for *augmint* processing.

Building with commit 97845578c6456b66a47b58dec9205b20d595cfe2

https://app.netlify.com/sites/augmint/deploys/5c828afdf726590008b12bde

**Comment by szerintedmi** [comment]
Looks really good! 

Couple of questions / suggestions:
- Any thoughts how it will work in conjunction with the delegated transfer? 
- copy should be shorter.  How about something like:
> **Care to help?**
> The recipient doesn't have ETH. They will need some to spend A-EUR. 
> Send some to them.
> 
- Change approx to ≈
- Why we don't have a default value entered? It's confusing that the tip in the field is an amount. I would have a default value there
- maybe remove the block after closing the rtanscation confirmation ? Or any scenario when would do it 2x?

The recipient/payee needs ETH to cover transaction fees, but does not have any... Care to help?
image

**Comment by admrid** [comment]
@szerintedmi i've done/changed all the things u mentioned.

regarding the delegated transfer: it seems that function is parked so i'll deal w it when it's coming

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "augmint-web",
-  "version": "0.1.0",
+  "version": "1.13.0",
```
Comment (phraktle) [inline package.json]: Is there a reason to maintain a version? If not, revert this change

File: src/containers/transfer/components/EthTransferForm.js [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+    }
+
+    onEthAmountChange(e) {
+        let amount;
+        try {
+            amount = parseFloat(e.target.value);
+            if (isNaN(amount)) {
+                amount = 0;
+            } else {
+                amount = parseFloat(e.target.value);
```
Comment (phraktle) [inline src/containers/transfer/components/EthTransferForm.js]: Redundant.

File: src/containers/transfer/components/EthTransferForm.js [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+            }
+        } catch (error) {
```
Comment (phraktle) [inline src/containers/transfer/components/EthTransferForm.js]: So is `parseFloat` supposed to return NaN or throw on invalid input? It should be one or the other, so drop the unused path.

File: src/containers/transfer/components/EthTransferForm.js [hunk 4]
```diff
@@ -0,0 +1,164 @@
...
+        return (
+            <div style={isFunctional && { display: "inline" }}>
+                {submitSucceeded && (
+                    <EthSubmissionSuccessPanel
+                        header="ETH transfer submitted"
+                        result={this.state.result}
+                        onDismiss={() => this.toggleEthTransferForm(false)}
+                    >
+                        <p>
+                            Transfer {this.state.ethAmount} ETH to {this.state.to}
```
Comment (phraktle) [inline src/containers/transfer/components/EthTransferForm.js]: Use `<ETH amount={…} />` component to display ETH amounts and drop the `.toFixed` call above.

File: src/containers/transfer/components/EthTransferForm.js [hunk 5]
```diff
@@ -0,0 +1,164 @@
...
+                        </p>
+                    </EthSubmissionSuccessPanel>
+                )}
+                {!submitSucceeded && (
+                    <Form
+                        error={error ? "true" : "false"}
```
Comment (phraktle) [inline src/containers/transfer/components/EthTransferForm.js]: Why not just `={error}`?

File: src/containers/transfer/components/TokenTransferForm.js [hunk 6]
```diff
@@ -1,8 +1,9 @@
 /*
+????
```
Comment (phraktle) [inline src/containers/transfer/components/TokenTransferForm.js]: ???? ?

File: src/modules/ethereum/transferTransactions.js [hunk 7]
```diff
@@ -42,6 +42,28 @@ export function getMaxTransfer(amount) {
     return maxAmount;
 }
 
+export async function transferEthTx(payload) {
+    const { payee, ethAmount } = payload;
+
+    const gasEstimate = cost.TRANSFER_AUGMINT_TOKEN_GAS; // TODO MAYBE change to & check ETH_GAS?
```
Comment (phraktle) [inline src/modules/ethereum/transferTransactions.js]: Indeed, that’s a very high estimate. Should use the default for transfers.

File: src/modules/payeeEthBalance.js [hunk 8]
```diff
@@ -0,0 +1,18 @@
...
+    const web3 = store.getState().web3Connect.web3Instance;
+
+    try {
+        const bn_weiBalance = await web3.eth.getBalance(payeeAddress);
+        console.log("Payee_Eth_Balance_Received");
+        return {
+            bn_ethBalance: bn_weiBalance,
+            ethBalance: web3.utils.fromWei(bn_weiBalance)
+        };
+    } catch (error) {
```
Comment (phraktle) [inline src/modules/payeeEthBalance.js]: Why not just throw?

File: src/modules/reducers/rootReducer.js [hunk 9]
```diff
@@ -22,6 +22,8 @@ import legacyLoanManagers from "modules/reducers/legacyLoanManagers";
 import stabilityBoardProxy from "modules/reducers/stabilityBoardProxy";
 import preToken from "modules/reducers/preToken";
 import metrics from "modules/reducers/metrics";
+// import payeeEthBalance from "modules/reducers/payeeEthBalance";
+// import ethTransfer from "modules/reducers/ethTransfer";
```
Comment (phraktle) [inline src/modules/reducers/rootReducer.js]: Remove.

File: src/modules/ethereum/transferTransactions.js (hunk 7 above)
Comment (szerintedmi) [reply src/modules/ethereum/transferTransactions.js]: shall we do it properly?
Add a new constant: `ETH_TRANSFER_GAS = 21000`

Comment (phraktle) [reply src/modules/ethereum/transferTransactions.js]: Doesn't web3 have a sensible default when unspecified? (Though a constant certainly works).

File: src/containers/transfer/components/EthTransferForm.js (hunk 4 above)
Comment (phraktle) [reply src/containers/transfer/components/EthTransferForm.js]: don't forget to remove `toFixed(2)`

Comment (admrid) [reply src/containers/transfer/components/EthTransferForm.js]: @phraktle  it seems i did not forget it, it is used for EUR on line 76

File: src/modules/payeeEthBalance.js (hunk 8 above)
Comment (phraktle) [reply src/modules/payeeEthBalance.js]: We'll just let this one slide for now... :)
